### PR TITLE
fix contact api throwing 500 when the data posted is invalid

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

**Comment by norkans7** [comment]
Fix for https://app.getsentry.com/nyaruka/rapidpro/group/49180567/

File: temba/api/serializers.py [hunk 1]
```diff
@@ -53,6 +53,15 @@ def from_native(self, data):
         else:
             raise ValidationError("Invalid, must be array: %s" % data)
 
+class WriteSerializer(serializers.Serializer):
+
+    def restore_fields(self, data, files):
+
+        if data is not None and not isinstance(data, dict):
+            self._errors['non_field_errors'] = ['Invalid data']
```
Comment (rowanseymour) [inline temba/api/serializers.py]: All of our write serializers expect a posted JSON object right? Then I think `data=None` should throw an error here, i.e just check `if not isinstance(data, dict)`.

Does creating an error here prevent further processing by the serializer?

Maybe also make error message more descriptive like *"Request body should be a single JSON object"*

Comment (norkans7) [inline temba/api/serializers.py]: RIght, I adjust that

Comment (norkans7) [inline temba/api/serializers.py]: we check if the validity (not serializer._errors) of a serializer before saving

**Pull request merged**